utils.go

```diff
@@ -260,6 +260,29 @@ func moveFile(sourcePath, destinationPath string) error {
 	return nil
 }
 
+func backupFile(fileName string) (newName string, err error) {
+	var backup func(f string) error
+	backup = func(f string) error {
+		newName = f + ".bak"
+		exists, err := fileExists(newName)
+		if err != nil {
+			return err
+		} else if exists {
+			return backup(newName)
+		}
+
+		err = moveFile(fileName, newName)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	err = backup(fileName)
+	return
```

Comment:
The nested function doesn't seem to be necessary. Can just be
```
	newName = filename + ".bak"
	exists, err := fileExists(newName)
	if err != nil {
		return err
	} else if exists {
		return backupFile(newName)
	}

	err = moveFile(fileName, newName)
	if err != nil {
		return err
	}

	return nil
```

Reply:
I added the nested function just so I could return the newName and log it to the user, this implementation ^^^ does not do that.

Reply:
I updated the code snippet.

Reply:
Updated